onadata/apps/api/tests/viewsets/test_data_viewset.py

```diff
@@ -2313,6 +2316,38 @@ def test_data_query_ornull(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(len(response.data), 4)
 
+    @patch(
+        'onadata.apps.api.viewsets.data_viewset.send_mqtt_message')
+    def test_send_mqtt_message_upon_deletion(self, send_message_mock):
+        self._make_submissions()
+        self.xform.refresh_from_db()
+        formid = self.xform.pk
+        initial_count = self.xform.instances.filter(deleted_at=None).count()
+        self.assertEqual(initial_count, 4)
+        self.assertEqual(self.xform.num_of_submissions, 4)
+        view = DataViewSet.as_view({'delete': 'destroy'})
+
+        # test with valid instance id's
+        records_to_be_deleted = self.xform.instances.all()[:2]
+        instance_ids = ','.join([str(i.pk) for i in records_to_be_deleted])
+        data = {"instance_ids": instance_ids}
+
+        request = self.factory.delete('/', data=data, **self.extra)
+        response = view(request, pk=formid)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data.get('message'),
+            "%d records were deleted" % len(records_to_be_deleted)
+        )
+        self.assertTrue(send_message_mock.called)
+
+        self.xform.refresh_from_db()
+        current_count = self.xform.instances.filter(deleted_at=None).count()
+        self.assertNotEqual(current_count, initial_count)
+        self.assertEqual(current_count, 2)
+        self.assertEqual(self.xform.num_of_submissions, 2)
+
```

Comment:
Instead of a new test - should we add the mock to the `test_deletion_of_bulk_submissions()` test and the call `self.assertTrue(send_message_mock.called)` where appropriate?  https://github.com/onaio/onadata/blob/ab883ce35d747bdb58da9e6b03b9a58abf9cc5a5/onadata/apps/api/tests/viewsets/test_data_viewset.py#L1413

My concern is additional - repeated test adds more time to running tests and I would hope we want shorter timelines with full test coverage.